app/controllers/businessdetails/ConfirmRegisteredOfficeController.scala

```diff
@@ -37,15 +37,26 @@ class ConfirmRegisteredOfficeController @Inject () (val dataCache: DataCacheConn
                                                     val ds: CommonPlayDependencies,
                                                     val cc: MessagesControllerComponents) extends AmlsBaseController(ds, cc) {
 
-
   def updateBMAddress(bm: BusinessMatching): Option[RegisteredOffice] = {
-    bm.reviewDetails.fold[Option[RegisteredOffice]](None)(dtls => Some(RegisteredOfficeUK(
-      dtls.businessAddress.line_1,
-      dtls.businessAddress.line_2,
-      dtls.businessAddress.line_3,
-      dtls.businessAddress.line_4,
-      dtls.businessAddress.postcode.getOrElse("")
-    )))
+    bm.reviewDetails.fold[Option[RegisteredOffice]](None)(dtls =>
+      if(dtls.businessAddress.postcode.isDefined) {
```

Comment:
Do we have a test that covers dtls being `None`? Just to make sure this line can handle that?

Reply:
Thanks, this was a valid point. Theoretically this should never happen that there is no review details but just in case, I've tightened condition on redirection to next page. Now if the address is none user will be redirected to address questions. There is also test for that.